Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift

```diff
@@ -117,6 +99,28 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
                 return try retryClose(db: db, exponentialBackoff: &exponentialBackoff)
             }
         }
+
+        // Signal long-running operation (e.g., populateTargetTrie) to stop
+        if case .connected(let db) = try self.withStateLock({ self.state }) {
+            try self.withStateLock {
+                self.state = .disconnecting(db)
+            }
+
+            do {
+                try db.close()
+            } catch {
+                do {
+                    var exponentialBackoff = ExponentialBackoff()
+                    try retryClose(db: db, exponentialBackoff: &exponentialBackoff)
+                } catch {
+                    throw StringError("Failed to close database")
+                }
+            }
+        }
+
+        try self.withStateLock {
```

Comment:
if I read this correctly this will set the state to disconnected always. intentional?

Reply:
Yes, this is the end of `close()` and when we get here it means we've closed the connection or gave up.